sys/kern/imgact_elf.c

```diff
@@ -574,12 +578,13 @@ __elfN(map_insert)(struct image_params *imgp, vm_map_t map, vm_object_t object,
 		    round_page(start), prot);
 		if (rv != KERN_SUCCESS)
 			return (rv);
-		offset += round_page(start) - start;
+		offset += (vaddr_t)round_page(start) - (vaddr_t)start;
```

Comment:
For now it is a uintcap_t arithmetic warning for the provenance, I also wish it wouldn't be necessary. Perhaps we could always interpret this as pointer difference, and whenever you want to keep a tagged value you have to be explicit about which one? Or maybe we can always interpret it as pointer difference if the result is being coerced into a non-capability type, although this might be confusing